microceph/ceph/config.go

```diff
@@ -3,15 +3,140 @@ package ceph
 import (
 	"context"
 	"database/sql"
+	"encoding/json"
 	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
 
+	"github.com/canonical/microceph/microceph/api/types"
 	"github.com/canonical/microceph/microceph/common"
 	"github.com/canonical/microceph/microceph/database"
 )
 
+type ConfigExtras struct{
+	Who     string // Ceph Config internal <who> against each key
+	Daemons []string // Daemons that need to be restarted post config change.
+	Regexp  string // Regular Expression to check value against key
+	Cb      *func(s common.StateInterface, c types.Config)
+}
+
+type ConfigTable map[string]ConfigExtras
+
+// Check if certain key is present in the map.
+func (c ConfigTable) isKeyPresent(key string) bool {
+	if _, ok := c[key]; !ok {
+        return false
+    }
+
+	return true
+}
+
+// Return keys of the given set
+func (c ConfigTable) Keys() (keys []string) {
+    for k := range c {
+        keys = append(keys, k)
+    }
+    return keys
+}
+
+func GetConfigTable() ConfigTable {
```

Comment:
By convention go constructors are named like `NewFoo()`, suggest to follow this here